src/batCharParser.ml

```diff
@@ -66,11 +66,11 @@ let string s = label ("\"" ^ s ^ "\"") (
   )
 
 let case_char c =
-  if BatChar.is_letter c then one_of [Char.uppercase c; Char.lowercase c]
+  if BatChar.is_letter c then one_of [Char.uppercase_ascii c; Char.lowercase_ascii c]
   else char c
 
 let case_string s = label ("case insensitive \"" ^ s ^ "\"") (
-    let s'  = String.lowercase s in
+    let s'  = String.lowercase_ascii s in
```

Comment:
Not sure if it matters to anyone, but `lowercase`→`lowercase_ascii`, etc slightly change behavior on OCaml < 5 as well, because the non-`_ascii` versions used Latin-1.

Reply:
Ok, so  a safer route would be to not change for OCaml < 5 (using preprocessing) but announce for ocaml5 a breaking change for people who were dealing with latin-1 strings.

Reply:
this is more conservative now (ocaml < 5) is not impacted by the change; using preprocessing